osu.Game/Rulesets/Objects/Legacy/ConvertHitObjectParser.cs

```diff
@@ -264,35 +312,36 @@ private PathControlPoint[] convertPathString(string pointString, Vector2 offset)
             while (++endIndex < pointSplit.Length)
             {
                 // Keep incrementing endIndex while it's not the start of a new segment (indicated by having a type descriptor of length 1).
-                if (pointSplit[endIndex].Length > 1)
+                if (pointString[pointSplit[endIndex]].Length > 1)
```

Comment:
Is it possible to have `pointSplit` be an array of `ReadOnlySpan<char>` rather than an array of `Range`s, to better mimic how a split operation would look? Because this is... headache territory (outer indexer is a string slice, inner indexer is access to array of ranges).

Reply:
`Span<Span>` is something on the wish list of C# and would require CLR work too, together with tuple of spans. `Span*` is a thing now but we can't alloc any array or stackalloc. This method is really the biggest headache in this PR. Since sliders are really common, fallback to string is likely to harm the performance.

Reply:
Another approach I have considered: parsing the whole point string into `PointOrControl[]` (a manually simulated union) in the caller, then the callee can deal it easily.